pretyping/typing.ml

```diff
@@ -534,8 +534,8 @@ let rec execute env sigma cstr =
 
     | Evar ev ->
         let ty = EConstr.existential_type sigma ev in
-        let sigma, jty = execute env sigma ty in
-        let sigma, jty = assumption_of_judgment env sigma jty in
+        let jty = Retyping.get_type_of env sigma ty in
```

Comment:
This is wrong, retyping assume the argument to be well-typed but here we precisely don't know about `ty` yet.

Reply:
The issue in `Case13.v` looks a lot like a bug in the compilation of matches. In `prepare_predicate_from_arsign_tycon`, the `let p = predicate 0 c in` line changes the argument of the evar `c` by something else, but this argument can appear in the type of other arguments which are not changed, so `p` is ill-typed.